lib/whitehall_importer/migrate_assets.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class MigrateAssets
+    attr_reader :whitehall_import
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(whitehall_import)
+      @whitehall_import = whitehall_import
+    end
+
+    def call
+      whitehall_import.migratable_assets.each do |whitehall_asset|
+        begin
+          migrate_asset(whitehall_asset)
+        rescue GdsApi::BaseError, StandardError => e
+          whitehall_asset.update!(state: "migration_failed", error_message: e.inspect)
+        end
+      end
+    end
+
+  private
+
+    def migrate_asset(whitehall_asset)
+      whitehall_asset.update!(state: "processing")
+      if whitehall_asset.content_publisher_asset.present? && whitehall_asset.content_publisher_asset.live?
```

Comment:
You can probably reduce the two conditions to: `whitehall_asset.content_publisher_asset.&live?`